src/main/java/graphql/normalized/ExecutableNormalizedField.java

```diff
@@ -61,20 +65,58 @@ private ExecutableNormalizedField(Builder builder) {
         this.parent = builder.parent;
     }
 
+    @Deprecated // doesn't work
     public boolean isConditional(GraphQLSchema schema) {
         if (parent == null) {
             return false;
         }
         return objectTypeNames.size() > 1 || unwrapAll(parent.getType(schema)) != getOneObjectType(schema);
     }
 
+    public boolean isConditional(GraphQLSchema schema, @Nullable String parentObjectTypeName) {
```

Comment:
Can we not work this out at ENF object creation time.  eg we have the parent field in scope as we create child ones.
So iis it more efficient to know the "parent is an interface" and hence the child field is conditional?

This then becomes a boolean without having to look upwards

Thoughts??

Reply:
We _could_ but Nadel needs to manipulate and insert new objects which would need it computed for them.

An optimization for later I'd say?